app.py

```diff
@@ -0,0 +1,142 @@
+from datetime import datetime
+from flask import Flask, render_template, request, request, jsonify
+from dotenv import load_dotenv
+
+#models
+from Infrastructure.domain.entities import models
+
+import os
+import requests
+import json
+
+load_dotenv()
+app = Flask(__name__)
+
+AUTHORIZE_URL = os.getenv("RSO_BASE_URI") + "/authorize"
+TOKEN_URL = os.getenv("RSO_BASE_URI") + "/token"
+LOGOUT_URL = os.getenv("RSO_BASE_URI") + "/logout"
+APP_CALLBACK_URL = os.getenv("APP_BASE_URL") +":"+ os.getenv("PORT") + os.getenv("APP_CALLBACK_PATH")
+RSO_CLIENT_ID = os.getenv("RSO_CLIENT_ID")
+AUTH_URL=AUTHORIZE_URL + "?redirect_uri=" + APP_CALLBACK_URL + "&client_id=" + RSO_CLIENT_ID + "&response_type=code" + "&scope=openid"
+token = ""
+
+def set_token(value: str):
+    global token
+    token = value
+
+@app.route('/', methods=['POST','GET'])
+def index(): 
+    return render_template('index.html', AUTH_URL = AUTH_URL, LOGOUT_URL=LOGOUT_URL)
+
+@app.route('/oauth-callback', methods=['POST','GET'])
+def aouthCallback():
+    code = request.args.get('code')
+
+    data = {
+        'grant_type': "authorization_code",
+        'code': code,
+        'redirect_uri': APP_CALLBACK_URL
+        }
+
+    try:
+        session = requests.Session()
+        session.auth = (os.getenv('RSO_CLIENT_ID'), os.getenv('RSO_CLIENT_SECRET'))
+
+        r = session.post(
+            url = TOKEN_URL, 
+            data = data, 
+            headers={"Content-Type":"application/x-www-form-urlencoded"},
+            )
+
+        set_token(r.json().get('access_token'))
+
+        return render_template('main.html', account={})   
+    except:
+        return render_template('index.html', AUTH_URL = AUTH_URL, LOGOUT_URL=LOGOUT_URL)   
+
+
+@app.route('/account', methods=['POST','GET'])
+def account():
+
+    if request.method == 'POST':
+        requestURL = "https://" + os.getenv('REGION') + ".api.riotgames.com/riot/account/v1/accounts/me"
+        header = {"Authorization": 'Bearer ' + token}
+
+        try:
+            response = requests.get(requestURL, headers=header)
+
+            if response.status_code != 200: 
+                return render_template('main.html', ACCOUNT_INFO = response.json().get('status')["message"])
+
+            
+            return render_template('main.html', ACCOUNT_INFO = response.json())
+        except:
+
+            return render_template('main.html', ACCOUNT_INFO = "Error getting account info ")
+
+
+@app.route('/accounts/me', methods=['GET'])
+def accountMe():
+
+    ## validate token 
+    if token != "": 
+
+        requestURL = "https://" + os.getenv('REGION') + ".api.riotgames.com/riot/account/v1/accounts/me"
+        header = {"Authorization": 'Bearer ' + token}
+
+        try:
+            response = requests.get(requestURL, headers=header)
+
+            if response.status_code != 200: 
+                return jsonify("Not data was found"), 401  
+
+            return response.json(), 200  
+        except:
```

Comment:
Should we be returning the exception, or at least logging it?

Reply:
updated